src/main/webapp/mapsScript.js

```diff
@@ -39,7 +39,7 @@ function initMap(centerLat, centerLng, zoomLevel = 12) {
 /** 
  * Adds the waypoint as markers to the map and draws a polyline between them to show the route.
  */
-function addMarkersToMap(waypoints) {
+function addMarkersAndDetailsToMap(waypoints) {
```

Comment:
We should have some tests for this, since there is real logic here, not just dom manipulation.